extensions/target-specific/chromium/compositorframe/compositorframe.js

```diff
@@ -0,0 +1,102 @@
+//--------------------------------------------------------------
+//
+//    MIT License
+//
+//    Copyright (c) Microsoft Corporation. All rights reserved.
+//
+//--------------------------------------------------------------
+
+"use strict";
+
+var CompositorFrame = undefined;
+Loader.OnLoad(function () {
+  CompositorFrame = {
+    Tree: new DbgObjectTree.DbgObjectTreeReader(),
+    Renderer: new DbgObjectTree.DbgObjectRenderer(),
+    InterpretAddress: function (address) {
+      var voidObj = DbgObject.create("ntdll!void", address);
+      if (!voidObj.isNull()) {
+        return voidObj.as(Chromium.GpuProcessType("viz::CompositorFrame"));
+      }
+      return DbgObject.NULL;
+    },
+    GetRoots: function () {
+      return DbgObject.locals("service", "viz::CompositorFrameSinkImpl::SubmitCompositorFrameInternal", "frame").then(
+        (frames) => {
+          return frames;
+        }
+      );
+    },
+    DefaultTypes: []
+  };
+
+  CompositorFrame.Tree.addChildren(Chromium.GpuProcessType("viz::CompositorFrame"), (parentLayer) => {
+    return parentLayer.f("render_pass_list").array("Elements").map(unique_ptr => { return unique_ptr.F("Object"); });
+  });
+
+  CompositorFrame.Tree.addChildren(Chromium.GpuProcessType("viz::RenderPass"), (parentLayer) => {
+    return Promise.all([parentLayer.f("quad_list").f("helper_").f("data_").F("Object").pointerValue()]).then(
+      (charAllocator) => {
+        var voidObj = DbgObject.create("ntdll!void**", charAllocator);
+        if (!voidObj.isNull()) {
+          // voidObj is a std::vector<std::unique_ptr<InnerList>>
+          return voidObj.size().then((platformPointerSize) => {
+            return voidObj.vals(2).then((val) => {
+              // Memory address from val[0] to val[1] contain pointers
+              var count = (val[1] - val[0]) / platformPointerSize;
+              var address = val[0];
+              var innerListContentsArrayPromises = [];
+              for (var i = 0; i < count; i++) {
+                var innerList = DbgObject.create("ntdll!void**", address);
+                // Inner list has 4 values of interest data,capacity,size,step.
+                innerListContentsArrayPromises.push(innerList.val().then((addrInsertList) => {
```

Comment:
Isn't innerList.val() just address?

Reply:
yes it is the address of the innerlist struct - whose symbol is not emitted in the pdb since it is defined in the .cc file. I cast to a void* and read the memory to get around this.

Reply:
No, I meant you don't have to get innerList.val(). You can just do this:

innerListContentsArrayPromises.push(DbgObject.create("ntdll!void**", address).vals(4))

Reply:
I tried this again now, results in 404.

Reply:
@sushraja-msft out of curiousity, if the type wasn't emitted in the PDB, what type did it get without the cast?